cmd/nodecmd/create.go

```diff
@@ -341,18 +500,70 @@ func createClusterNodeConfig(network models.Network, cloudConfigMap models.Cloud
 			if err != nil {
 				return err
 			}
-			if err = addNodeToClustersConfig(network, cloudConfig.InstanceIDs[i], clusterName); err != nil {
-				return err
-			}
-			if err := updateKeyPairClustersConfig(cloudConfigMap[region]); err != nil {
+			if err = addNodeToClustersConfig(network, cloudConfig.InstanceIDs[i], clusterName, false); err != nil {
 				return err
 			}
 		}
+		publicIP := ""
+		if useStaticIP {
+			publicIP = monitorCloudConfig.PublicIPs[0]
+		}
+		nodeConfig := models.NodeConfig{
+			NodeID:        monitorCloudConfig.InstanceIDs[0],
+			Region:        monitoringHostRegion,
+			AMI:           monitorCloudConfig.ImageID,
+			KeyPair:       monitorCloudConfig.KeyPair,
+			CertPath:      monitorCloudConfig.CertFilePath,
+			SecurityGroup: monitorCloudConfig.SecurityGroup,
+			ElasticIP:     publicIP,
+			CloudService:  cloudService,
+		}
+		if err := app.CreateNodeCloudConfigFile(monitorCloudConfig.InstanceIDs[0], &nodeConfig); err != nil {
+			return err
+		}
+		if err := addNodeToClustersConfig(network, monitorCloudConfig.InstanceIDs[0], clusterName, true); err != nil {
+			return err
+		}
+		if err := updateKeyPairClustersConfig(nodeConfig); err != nil {
+			return err
+		}
 	}
 	return nil
 }
 
-func updateKeyPairClustersConfig(cloudConfig models.RegionConfig) error {
+func addHTTPHostToConfigFile(filePath string) error {
+	jsonFile, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer jsonFile.Close()
+	byteValue, _ := io.ReadAll(jsonFile)
+	var result map[string]interface{}
+	if err := json.Unmarshal(byteValue, &result); err != nil {
+		return err
+	}
+	result["http-host"] = "0.0.0.0"
+	byteValue, err = json.MarshalIndent(result, "", "    ")
+	if err != nil {
+		return err
+	}
+	return os.WriteFile(filePath, byteValue, constants.WriteReadReadPerms)
+}
+
+func getExistingMonitoringInstance(clusterName string) (string, error) {
+	if app.ClustersConfigExists() {
+		clustersConfig, err := app.LoadClustersConfig()
+		if err != nil {
+			return "", err
+		}
+		if clustersConfig.Clusters[clusterName].MonitoringInstance != "" {
```

Comment:
check for valid clusterName is missing. 
without it and with wrong name it will crash